docs/reference/updating-mapper.md

```diff
@@ -0,0 +1,42 @@
+# Updating the doc mapping of an index
+
+Quickwit allows updating the mapping it uses, to add more fields to an existing index, or change how they are indexed. In doing so,
+it does not reindex existing data, still lets you search through older documents where possible.
+
+## Indexing
+
+When receiving a new doc mapping for an index, Quickwit will restart indexing pipelines to take into account the changes. This operation
+is asynchronous. You can't assume documents send immediately after the update will necessarily use the new mapping. On the other hand,
+this usually happens very fast, possibly while some documents sent before the udpate haven't been commited yet. This means you can't
+assume documents sent before the update won't be indexed using the new doc mapping.
+
+## Querying
+
+When receiving a query, Quickwit always validate that query against the most recent mapping. If a query was valid under a previous
+mapping but is not compatible with the newer mapping, that query will be rejected. For instance if a field which was indexed no longer
+is, or a field was removed, any query that uses it will become invalid. On the other hand, if a query was not valid for a previous doc
+mapping, but is valid under the new doc mapping, Quickwit will process the query. When querying newer splits, it will behave normally,
+when querying older splits, it will try ot execute the query as correctly as possible. If you find a situation where older splits
+causes a valid request to return an error, please open a bug report.
+
+For instance, if a field was added or newly marked as indexed, quickwit will behave as if no older document had the right value. This
```

Comment:
It would be nice to have a table to describe the change in one column and the behavior in a second column.